src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Manufacturer/Address/Blocks/form.html.twig

```diff
@@ -72,7 +72,7 @@
           'label': 'Country'|trans({}, 'Admin.Global'),
       }) }}
 
-      <div class="js-manufacturer-address-state">
+      <div class="js-manufacturer-address-state {{ addressForm.children.id_state.vars.attr.class }}">
```

Comment:
Would it be possible to avoid front-end classes coming from the back ? Like this :
```suggestion
      <div class="js-manufacturer-address-state{% if addressForm.children.id_state.vars.isHidden %} d-none{% endif %}">
```

Reply:
https://github.com/PrestaShop/PrestaShop/pull/16938/files#diff-bd2da52b9459a118b619e517064ca188R431
It seems that with new form rendering we are still doing all of that in backend (even tho I agree it doesn't sound right) :man_shrugging: .
Another way would be to pass vars from controller to twig, but thats an overhead.
And yet another way is to pass Hidden attribute, but its also kinda same as class, just this case would require even more code to make js toggle component work.

Reply:
It's not as same as a class,  because it means we can still modify the class and css linked to it without going to backend, while passing a complete class name force you to work on the backend if you want to modify the class. Even if we already did it somewhere, it doesn't mean that it make sense 😄 

We should try to separate as much as possible the front-end job from the backend one, as we are thinking of a transition to VueJS, this is such a thing we have to avoid absolutely in order to split as much as possible the back from the front event if we're on a monolithic application.

Also an other point it that we don't know at all what that `{{ addressForm.children.id_state.vars.attr.class }}` does while reading the twig, but with `{% if addressForm.children.id_state.vars.isHidden %} d-none{% endif %}` we instantly know what this does, it's just a question about splitting backend logic and front-end one :)

Reply:
@PrestaShop/prestashop-core-developers wdyt ?

Reply:
so your final suggestion is to try work it out with hidden attribute? This would mean some tweaking in js component (but shouldn't be very hard) ?